packages/astro/src/@types/astro.ts

```diff
@@ -716,6 +695,16 @@ export interface AstroUserConfig {
 	buildOptions?: never;
 	/** @deprecated `devOptions` has been renamed to `server` */
 	devOptions?: never;
+
+	legacy?: {
+		/**
+		 * Enable components and JSX expressions in markdown
+		 * Consider our MDX integration before applying this flag!
+		 * @see https://docs.astro.build/en/guides/integrations-guide/mdx/
+		 * Default: false
+		 */
+		 astroFlavoredMarkdown?: boolean;
```

Comment:
@tony-sull If you want `legacy.astroFlavoredMarkdown` to show up in docs, then I think you need to put your content here.